src/main/java/com/google/devtools/build/lib/rules/cpp/ShowIncludesFilter.java

```diff
@@ -54,10 +56,37 @@ public ShowIncludesFilter(String sourceFileName, String workspaceName) {
     private final ByteArrayOutputStream buffer = new ByteArrayOutputStream(4096);
     private final Collection<String> dependencies = new ArrayList<>();
     private static final int NEWLINE = '\n';
-    private static final String SHOW_INCLUDES_PREFIX = "Note: including file:";
+    // "Note: including file:" in 14 languages,
+    // cl.exe will print different prefix according to the locale configured for MSVC.
+    private static final List<String> SHOW_INCLUDES_PREFIXS = Arrays.asList(
+        HexToUTF8String("4E6F74653A20696E636C7564696E672066696C653A"), // English
```

Comment:
These are byte arrays as string.
Conversion costs time, and we lack compile-time check for ranges.
Please convert these strings to byte arrays of the UTF-8 data.

Reply:
This is a static member of the class, so they will only be converted for once. I think it won't cause any performance regression and it's more clean to write as string than an array of integers (I didn't find a way to write a list of byte constant).

What do you mean by "we lack compile-time check for ranges"?

Reply:
Readability suffers already: it's a byte stream either way -- now it's a string of hex values, otherwise it would be a byte array. Lack of compile time check: we might add a non-hexadecimal character in the string, and we'd only know at runtime.

How about this:

```java
Utf8BytesToString(new byte[] { 0xe6, 0xb3, 0xa8, 0xe6, 0x84, 0x8f, 0x3a, 0x20, 0xe5, 0x8c, 0x85, 0xe5, 0x90, 0xab, 0xe6, 0xaa, 0x94, 0xe6, 0xa1, 0x88, 0x3a}),
Utf8BytesToString(new byte[] { 0x50, 0x6f, 0x7a, 0x6e, 0xc3, 0xa1, 0x6d, 0x6b, 0x61, 0x3a, 0x20, 0x56, 0xc4, 0x8d, 0x65, 0x74, 0x6e, 0xc4, 0x9b, 0x20, 0x73, 0x6f, 0x75, 0x62, 0x6f, 0x72, 0x75, 0x3a}),
Utf8BytesToString(new byte[] { 0x48, 0x69, 0x6e, 0x77, 0x65, 0x69, 0x73, 0x3a, 0x20, 0x45, 0x69, 0x6e, 0x6c, 0x65, 0x73, 0x65, 0x6e, 0x20, 0x64, 0x65, 0x72, 0x20, 0x44, 0x61, 0x74, 0x65, 0x69, 0x3a}),
Utf8BytesToString(new byte[] { 0x4e, 0x6f, 0x74, 0x65, 0x3a, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x64, 0x69, 0x6e, 0x67, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x3a}),
Utf8BytesToString(new byte[] { 0x52, 0x65, 0x6d, 0x61, 0x72, 0x71, 0x75, 0x65, 0xc2, 0xa0, 0x3a, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x73, 0x69, 0x6f, 0x6e, 0x20, 0x64, 0x75, 0x20, 0x66, 0x69, 0x63, 0x68, 0x69, 0x65, 0x72, 0xc2, 0xa0, 0x3a}),
Utf8BytesToString(new byte[] { 0x4e, 0x6f, 0x74, 0x61, 0x3a, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x73, 0x6f}),
Utf8BytesToString(new byte[] { 0xe3, 0x83, 0xa1, 0xe3, 0x83, 0xa2, 0x3a, 0x20, 0xe3, 0x82, 0xa4, 0xe3, 0x83, 0xb3, 0xe3, 0x82, 0xaf, 0xe3, 0x83, 0xab, 0xe3, 0x83, 0xbc, 0xe3, 0x83, 0x89, 0x20, 0xe3, 0x83, 0x95, 0xe3, 0x82, 0xa1, 0xe3, 0x82, 0xa4, 0xe3, 0x83, 0xab, 0x3a}),
Utf8BytesToString(new byte[] { 0xec, 0xb0, 0xb8, 0xea, 0xb3, 0xa0, 0x3a, 0x20, 0xed, 0x8f, 0xac, 0xed, 0x95, 0xa8, 0x20, 0xed, 0x8c, 0x8c, 0xec, 0x9d, 0xbc, 0x3a}),
Utf8BytesToString(new byte[] { 0x55, 0x77, 0x61, 0x67, 0x61, 0x3a, 0x20, 0x77, 0x20, 0x74, 0x79, 0x6d, 0x20, 0x70, 0x6c, 0x69, 0x6b, 0x75, 0x3a}),
Utf8BytesToString(new byte[] { 0x4f, 0x62, 0x73, 0x65, 0x72, 0x76, 0x61, 0xc3, 0xa7, 0xc3, 0xa3, 0x6f, 0x3a, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x69, 0x6e, 0x64, 0x6f, 0x20, 0x61, 0x72, 0x71, 0x75, 0x69, 0x76, 0x6f, 0x3a}),
Utf8BytesToString(new byte[] { 0xd0, 0x9f, 0xd1, 0x80, 0xd0, 0xb8, 0xd0, 0xbc, 0xd0, 0xb5, 0xd1, 0x87, 0xd0, 0xb0, 0xd0, 0xbd, 0xd0, 0xb8, 0xd0, 0xb5, 0x3a, 0x20, 0xd0, 0xb2, 0xd0, 0xba, 0xd0, 0xbb, 0xd1, 0x8e, 0xd1, 0x87, 0xd0, 0xb5, 0xd0, 0xbd, 0xd0, 0xb8, 0xd0, 0xb5, 0x20, 0xd1, 0x84, 0xd0, 0xb0, 0xd0, 0xb9, 0xd0, 0xbb, 0xd0, 0xb0, 0x3a}),
Utf8BytesToString(new byte[] { 0x4e, 0x6f, 0x74, 0x3a, 0x20, 0x65, 0x6b, 0x6c, 0x65, 0x6e, 0x65, 0x6e, 0x20, 0x64, 0x6f, 0x73, 0x79, 0x61, 0x3a}),
Utf8BytesToString(new byte[] { 0xe6, 0xb3, 0xa8, 0xe6, 0x84, 0x8f, 0x3a, 0x20, 0xe5, 0x8c, 0x85, 0xe5, 0x90, 0xab, 0xe6, 0x96, 0x87, 0xe4, 0xbb, 0xb6, 0x3a}),
Utf8BytesToString(new byte[] { 0x4e, 0x6f, 0x74, 0x61, 0x3a, 0x20, 0x69, 0x6e, 0x63, 0x6c, 0x75, 0x73, 0x69, 0xc3, 0xb3, 0x6e, 0x20, 0x64, 0x65, 0x6c, 0x20, 0x61, 0x72, 0x63, 0x68, 0x69, 0x76, 0x6f, 0x3a}),
```

Reply:
Thanks! But I had to use `new int[]`, otherwise I got
```
error: incompatible types: possible lossy conversion from int to byte
```

Reply:
Ah, because all Java's number types are signed. Sheesh.